packages/flame/lib/src/components/component.dart

```diff
@@ -203,6 +206,16 @@ class Component {
   /// ```
   void onMount() {}
 
+  /// A future that will complete once the component is mounted on its parent
+  ///
+  /// Note: Calling this method after the component has already mounted will
```

Comment:
@erickzanardo I wonder if it could be possible to avoid this being a never-ending future when it has already mounted.

If I were using this API and read `hasMounted` I'll expect to receive a boolean value from it. I wonder if `whenMounted` or `onMounted` is a better name for it. This leads me to think that perhaps passing a callback to these is a more intuitive API (reminds me of hooks).  Something as `whenMounted(() => print('hello'))`.

In the case that the `Component` has already been mounted I think I'll expect the callback to never be reached.

wdyt?

Reply:
I don't know which way I prefer. 🤔 
For collision detection we have just a nullable field where you can set the wanted callback, that could be an option too.

Reply:
Actually, this is very easy to do, I can just check if it is already mounted, and then just return a `Future.value` instead of the future of the completer, I am gonna change this